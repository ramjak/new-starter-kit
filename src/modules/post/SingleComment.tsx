import React from 'react';
import IComment from '../../domains/comment';

interface ISingleComment {
  comment: IComment;
}

function SingleComment({ comment }: ISingleComment) {
  return (
    <>
      <article>
        <p>{comment.body}</p>
        <address>
          By:{' '}
          <a href={`mailto:${comment.email}`} rel="author">
            {comment.name}
          </a>
        </address>
      </article>
      <hr />
    </>
  );
}

export default SingleComment;
