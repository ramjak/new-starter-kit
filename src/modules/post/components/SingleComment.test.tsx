import "reflect-metadata";
import React from "react";
import renderer from "react-test-renderer";
import SingleComment from "./SingleComment";

it("render SingleComment", () => {
  const result = renderer
    .create(
      <SingleComment
        comment={{
          id: 1,
          body:
            "laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium",
          postId: 2,
          name: "poster",
          email: "test@email.com",
        }}
      />
    )
    .toJSON();

  expect(result).toMatchSnapshot();
});
