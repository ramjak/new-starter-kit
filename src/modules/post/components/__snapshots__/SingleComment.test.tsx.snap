// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render SingleComment 1`] = `
Array [
  <article>
    <p>
      laudantium enim quasi est quidem magnam voluptate ipsam eos tempora quo necessitatibus dolor quam autem quasi reiciendis et nam sapiente accusantium
    </p>
    <address>
      By:
       
      <a
        href="mailto:test@email.com"
        rel="author"
      >
        poster
      </a>
    </address>
  </article>,
  <hr />,
]
`;
